### PR TITLE
refactor attendance

### PR DESCRIPTION
**Pull request by mariamisa**
relates #144

### CODE REVIEW

File: server/controller/routes/portal/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const {
-  validateAttendence,
+  validateAttendance: validateAttendance,
```
Comment (AhmedSafi97) [inline server/controller/routes/portal/index.js:2]:
```suggestion
  validateAttendance,
```

File: server/middleware/checkEmail.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const checkEmailIfExist = (req, res, next) => {
         next();
       } else {
         const error = new Error();
-        error.msg = 'email dosent exist';
+        error.msg = 'email docent exist';
```
Comment (AhmedSafi97) [inline server/middleware/checkEmail.js:13]:
```suggestion
        error.msg = 'email doesn't exist';
```

File: server/controller/routes/portal/attendance.js [hunk 3]
```diff
@@ -4,55 +4,65 @@ const { getEventDetails } = require('../../../database/queries/events');
...
-            "user hasn't booked this event yet, please book the event then try again",
-        });
-      }
-    })
-    .catch(next);
+const checkUserBooking = async (req, res, next) => {
+  const {
+    body: { userCode, eventId },
+  } = req;
+  try {
```
Comment (AlaaTaima) [inline server/controller/routes/portal/attendance.js:37]: `eventId` isn't within `req.body` , it comes from the previous middleware so, edit it like this : 
 ```suggestion
  const {
    body: { userCode},  eventId ,
  } = req;
  try {
```

File: server/controller/routes/portal/attendance.js [hunk 4]
```diff
@@ -4,55 +4,65 @@ const { getEventDetails } = require('../../../database/queries/events');
...
 };
 
-const signAttendance = (req, res, next) => {
-  signAttendanceSql(req.body.userCode, req.eventId)
-    .then(() => res.json({ msg: 'thanks for attending' }))
-    .catch(next);
+const signAttendance = async (req, res, next) => {
+  const {
+    body: { userCode, eventId },
+  } = req;
```
Comment (AlaaTaima) [inline server/controller/routes/portal/attendance.js:55]: edit it as the previous comment : 
```suggestion
  const {
    body: { userCode }, eventId ,
  } = req;
```

**Pull request merged**